nomad/state/state_store.go

```diff
@@ -3187,16 +3216,16 @@ func (s *StateStore) EvalsByJob(ws memdb.WatchSet, namespace, jobID string) ([]*
 
 // Evals returns an iterator over all the evaluations in ascending or descending
 // order of CreationIndex as determined by the ascending parameter.
-func (s *StateStore) Evals(ws memdb.WatchSet, ascending bool) (memdb.ResultIterator, error) {
+func (s *StateStore) Evals(ws memdb.WatchSet, reverse bool) (memdb.ResultIterator, error) {
```

Comment:
I called it `SortedDefault` because just `store.Evals(ws, state.Sorted)` felt like there would be a `Unsorted` option 😅 

I also didn't create a custom type because this is just a boolean that would often be loaded from somewhere else (like a request). So it felt a bit much to require type casting.

Reply:
Not to nitpick, but I do hate bool parameters as you can rarely tell what they're for at the call site. Here's an alternative similar to Tim's if you still have energy left to mess with it:

```go
type SortOrder bool
const SortAscending  SortOrder = true
const SortDescending SortOder = false

func (s *StateStore) Eval(ws memdb.WatchSet, sort SortOrder) (memdb.ResultIterator, error) { /* ... */ }

func SomeCallSite() {
  iter, err := state.Eval(ws, SortAscending)
}
```

Reply:
Yeah, that's what I started doing. I can follow through with that.

I also notice while writing the docs that it's super confusing that `reverse` only works with certain combination of query params, so I'm going to make it work in all cases. The good thing about renaming it to `reverse` is that it just means using `GetReverse` instead of `Get`, no need to mess with the index schema to get some kind of specific order out 🙂

Reply:
Actually, scratch that last part. It would add a lot of extra code (mainly tests, but still) to this PR that is already pretty big. I will raise smaller follow-up PRs for each endpoint.

@tgross @schmichael I used a named type value for setting the order in https://github.com/hashicorp/nomad/pull/12186/commits/6784b79e337fbe172eb09659088bbba0db03bd8d if you would like to take another look.

Reply:
ugh..sorry for the early ping, I'm working on fixing the tests.